pkg/flowkit/flowkit_test.go

```diff
@@ -0,0 +1,2045 @@
+		require.NoError(t, err)
+
+		tx, txr, err := flowkit.SendTransaction(
+			ctx,
+			roles,
+			NewScript(tests.TransactionSingleAuth.Source, nil, tests.TransactionSingleAuth.Filename),
+			flow.DefaultTransactionGasLimit,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, tx.Payer.String(), b.Address().String())
+		assert.Equal(t, tx.Authorizers[0].String(), c.Address().String())
+		assert.Equal(t, tx.ProposalKey.KeyIndex, a.Key().Index())
+		assert.Nil(t, txr.Error)
+		assert.Equal(t, txr.Status, flow.TransactionStatusSealed)
+	})
+
+	t.Run("Send Transaction two account roles", func(t *testing.T) {
+		t.Parallel()
+		state, flowkit := setupIntegration()
+		setupAccounts(state, flowkit)
+
+		a, _ := state.Accounts().ByName("Alice")
+		b, _ := state.Accounts().ByName("Bob")
+
+		roles, err := NewTransactionAccountRoles(a, b, []*Account{a})
+		require.NoError(t, err)
+
+		tx, txr, err := flowkit.SendTransaction(
+			ctx,
+			roles,
+			NewScript(tests.TransactionSingleAuth.Source, nil, tests.TransactionSingleAuth.Filename),
+			flow.DefaultTransactionGasLimit,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, tx.Payer.String(), b.Address().String())
+		assert.Equal(t, tx.Authorizers[0].String(), a.Address().String())
+		assert.Equal(t, tx.ProposalKey.KeyIndex, a.Key().Index())
+		assert.Nil(t, txr.Error)
+		assert.Equal(t, txr.Status, flow.TransactionStatusSealed)
+	})
+
+	t.Run("Send Transaction with arguments", func(t *testing.T) {
+		t.Parallel()
+		state, flowkit := setupIntegration()
+		setupAccounts(state, flowkit)
+
+		a, _ := state.Accounts().ByName("Alice")
+
+		tx, txr, err := flowkit.SendTransaction(
+			ctx,
+			NewSingleTransactionAccount(a),
+			NewScript(
+				tests.TransactionArgString.Source,
+				[]cadence.Value{
+					cadence.String("Bar"),
+				},
+				tests.TransactionArgString.Filename,
+			),
+			flow.DefaultTransactionGasLimit,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, tx.Payer.String(), a.Address().String())
+		assert.Equal(t, tx.ProposalKey.KeyIndex, a.Key().Index())
+		assert.Equal(t, fmt.Sprintf("%x", tx.Arguments), "[7b2276616c7565223a22426172222c2274797065223a22537472696e67227d]")
+		assert.Nil(t, txr.Error)
+		assert.Equal(t, txr.Status, flow.TransactionStatusSealed)
+	})
+
+	t.Run("Send transaction with multiple func declaration", func(t *testing.T) {
+		t.Parallel()
+		state, flowkit := setupIntegration()
+		setupAccounts(state, flowkit)
+
+		a, _ := state.Accounts().ByName("Alice")
+
+		tx, txr, err := flowkit.SendTransaction(
+			ctx,
+			NewSingleTransactionAccount(a),
+			NewScript(
+				tests.TransactionMultipleDeclarations.Source,
+				nil,
+				tests.TransactionMultipleDeclarations.Filename,
+			),
+			flow.DefaultTransactionGasLimit,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, tx.Payer.String(), a.Address().String())
+		assert.Equal(t, tx.ProposalKey.KeyIndex, a.Key().Index())
+		assert.Nil(t, txr.Error)
+		assert.Equal(t, txr.Status, flow.TransactionStatusSealed)
+	})
+}
```

Comment:
Might be nice to split these tests up a bit instead of one huge file in the future.

Reply:
I plan to refactor tests too, but separate PR